src/encoded/static/components/static-pages/components/StatisticsPageViewBody.js

```diff
@@ -228,63 +228,94 @@ export const commonParsingFxn = {
+                        'term': term,
+                        'count': trackingItemItem[countKey],
+                        'total': trackingItemItem[countKey],
+                        'date': for_date
                     };
+                    termsInAllItems.add(term);
+                    termsInCurrenItem.add(term);
+                }
+                totalSessions += trackingItemItem[countKey]
+                return { groupedTermsObj, totalSessions };
+            }, { groupedTermsObj: {}, totalSessions: 0 });
+            
+            totalSessionsToDate += totalSessions;
+
+            const currentItem = {
+                'date'      : for_date,
+                'count'     : cumulativeSum ? totalSessionsToDate : totalSessions,
+                'total'     : cumulativeSum ? totalSessionsToDate : totalSessions,
+                'children': _.values(groupedTermsObj).map(function (termItem) {
+                    const cloned = { ...termItem };
+                    if (cumulativeSum) {
+                        let { count = 0, total = 0 } = termTotals[termItem.term] || {};
+                        total += (termItem.total || 0);
+                        count += (termItem.count || 0);
+                        termTotals[termItem.term] = { total, count };
+
+                        cloned.count = count;
+                        cloned.total = total;
+                    }
+                    return cloned;
                 })
             };
 
-            // Unique-fy
-            currItem.children = _.values(_.reduce(currItem.children || [], function(memo, child){
-                if (memo[child.term]) {
-                    memo[child.term].count += child.count;
-                    memo[child.term].total += child.total;
-                } else {
-                    memo[child.term] = child;
-                }
-                return memo;
-            }, {}));
+            // add missing children for cumulative view
+            if (cumulativeSum) {
+                termsInAllItems.forEach(term => {
+                    if (!termsInCurrenItem.has(term)) {
+                        currentItem.children.push({
+                            'term': term,
+                            'count': termTotals[term].count,
+                            'total': termTotals[term].total,
+                            'date': for_date
+                        });
+                    }
+                });
+            }
 
             if (typeof topCount === 'number' && topCount > 0) {
-                currItem.children = _.sortBy(currItem.children, (item) => -1 * item.total).slice(0, topCount);
+                currentItem.children = _.sortBy(currentItem.children, (item) => -1 * item.total).slice(0, topCount);
             }
 
-            return currItem;
+            return currentItem;
 
-        }).reverse(); // We get these in decrementing order from back-end
+        }); // We get these in decrementing order from back-end
```

Comment:
Is this comment still true?

Reply:
yeah, definitely should not be there anymore!